### PR TITLE
Update deploy.yaml for OIDC

### PR DESCRIPTION
**Pull request by IAlibay**

### CODE REVIEW

**Comment by IAlibay** [comment]
Trusted publishing is now set up.

**Comment by IAlibay** [comment]
Also includes short term fix from: https://github.com/MDAnalysis/cookiecutter-mdakit/pull/93

**Comment by IAlibay** [comment]
Sorry to ping you again for review @orbeckst but at this point you've reviewed nearly all of them 🙃

**Comment by orbeckst** [comment]
That's ok, at least I have some idea what to expect.

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -9,29 +9,46 @@ on:
...
+    steps:
+      - uses: actions/checkout@v4
+
       - name: pypi_deploy
         uses: MDAnalysis/pypi-deployment@main
-        if: github.event_name == 'release' && github.event.action == 'published'
         with:
-          token: ${{ secrets.PYPI_API_TOKEN }}
           package_name: 'waterdynamics'
-          tests: false
```
Comment (orbeckst) [inline .github/workflows/deploy.yaml:37]: Is the default true or false? Maybe just be explicit because I saw that for test_pypi you are using `true`.

**Review by orbeckst — APPROVED** [review]

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (IAlibay) [reply .github/workflows/deploy.yaml:37]: yes but you are right, if we define it above we should define it here too!

**Pull request merged**